components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -8,15 +8,309 @@
+  base::StringPiece info("ethwallet");
+  return crypto::HkdfSha256(base::StringPiece(seed.begin(), seed.end()),
+      salt, info, kSeedByteLength);
+}
+
+// static
+bool BraveWalletService::LoadFromPrefs(PrefService* prefs,
+    std::string* cipher_seed, std::string* nonce) {
+  if (!prefs->HasPrefPath(kBraveWalletAES256GCMSivNonce) ||
+      !prefs->HasPrefPath(kBraveWalletEncryptedSeed)) {
+    return false;
+  }
+  if (!base::Base64Decode(
+          prefs->GetString(kBraveWalletAES256GCMSivNonce),
+          nonce)) {
+    return false;
+  }
+  if (!base::Base64Decode(
+          prefs->GetString(kBraveWalletEncryptedSeed),
+          cipher_seed)) {
+    return false;
+  }
+  return true;
+}
+
+// static
+bool BraveWalletService::OpenSeed(const std::string& cipher_seed,
+    const std::string& key, const std::string& nonce,
+    std::string* seed) {
+  crypto::Aead aes_256_gcm_siv(crypto::Aead::AES_256_GCM_SIV);
+  aes_256_gcm_siv.Init(&key);
+  return aes_256_gcm_siv.Open(cipher_seed, nonce, base::StringPiece(""), seed);
+}
+
+// Generate a new random nonce
+// static
+std::string BraveWalletService::GetRandomNonce() {
+  // crypto::RandBytes is fail safe.
+  uint8_t nonceBytes[kNonceByteLength];
+  crypto::RandBytes(nonceBytes, kNonceByteLength);
+  return std::string(
+      reinterpret_cast<char*>(nonceBytes), kNonceByteLength);
+}
+
+// static
+std::string BraveWalletService::GetRandomSeed() {
+  // crypto::RandBytes is fail safe.
+  uint8_t random_seed_bytes[kSeedByteLength];
+  crypto::RandBytes(random_seed_bytes, kSeedByteLength);
+  return std::string(
+      reinterpret_cast<char*>(random_seed_bytes), kSeedByteLength);
+}
+
+// static
+bool BraveWalletService::SealSeed(const std::string& seed,
+    const std::string& key, const std::string& nonce,
+    std::string* cipher_seed) {
+  crypto::Aead aes_256_gcm_siv(crypto::Aead::AES_256_GCM_SIV);
+  aes_256_gcm_siv.Init(&key);
+  return aes_256_gcm_siv.Seal(base::StringPiece(seed.begin(), seed.end()),
+      nonce, base::StringPiece(""), cipher_seed);
+}
+
+// Store the seed in preferences, binary pref strings need to be
+// base64 encoded.  Base64 encoding is fail safe.
+// static
+void BraveWalletService::SaveToPrefs(
+    PrefService* prefs,
+    const std::string& cipher_seed,
+    const std::string& nonce) {
+  // Store the seed in preferences, binary pref strings need to be
+  // base64 encoded.  Base64 encoding is fail safe.
+  std::string base64_nonce;
+  std::string base64_cipher_seed;
+  base::Base64Encode(nonce, &base64_nonce);
+  base::Base64Encode(base::StringPiece(cipher_seed.begin(),
+                                      cipher_seed.end()),
+      &base64_cipher_seed);
+  prefs->SetString(kBraveWalletAES256GCMSivNonce, base64_nonce);
+  prefs->SetString(kBraveWalletEncryptedSeed,
+      base64_cipher_seed);
+}
+
+void BraveWalletService::ResetCryptoWallets(
+    const base::FilePath& profile_path) {
+  const base::FilePath wallet_data_path = profile_path
```

Comment:
I know this is copied from the original code, but I think we need to address this in a follow-up because we can't just delete the extension settings out from under the extension system. The settings should be cleared by the extension itself. The only reason this even works is because we restart right afterwards, but it's still dangerous imo and I think it could potentially cause a crash before the restart if you got unlucky

Reply:
or if not the extension, it should be handled in the c++ code for the extension prefs - `ExtensionPrefs::Get(browser_context)->DeleteExtensionPrefs(extension_id);`

Reply:
Posted a follow up here https://github.com/brave/brave-browser/issues/8747 since it isn't related to this PR at all.